opendrift/export/io_netcdf.py

```diff
@@ -3,235 +3,82 @@
-    self.outfile.variables['trajectory'].units = '1'
-
-    self.outfile.Conventions = 'CF-1.11, ACDD-1.3'
-    self.outfile.standard_name_vocabulary = 'CF Standard Name Table v85'
-    self.outfile.featureType = 'trajectory'
-    self.outfile.title = 'OpenDrift trajectory simulation'
-    self.outfile.summary = 'Output from simulation with OpenDrift framework'
-    self.outfile.keywords = 'trajectory, drift, lagrangian, simulation'
-    self.outfile.history = 'Created ' + str(datetime.now())
-    self.outfile.date_created = datetime.now().isoformat()
-    self.outfile.source = 'Output from simulation with OpenDrift'
-    self.outfile.model_url = 'https://github.com/OpenDrift/opendrift'
-    self.outfile.opendrift_class = self.__class__.__name__
-    self.outfile.opendrift_module = self.__class__.__module__
-    self.outfile.readers = str(self.env.readers.keys())
-    self.outfile.time_coverage_start = str(self.start_time)
-    self.outfile.time_step_calculation = str(self.time_step)
-    self.outfile.time_step_output = str(self.time_step_output)
-
-    # Time
-    self.timeStr = 'seconds since 1970-01-01 00:00:00'
-    self.outfile.variables['time'].units = self.timeStr
-    self.outfile.variables['time'].standard_name = 'time'
-    self.outfile.variables['time'].long_name = 'time'
-    # Apparently axis attribute shall not be given for time, lon and lat
-    #self.outfile.variables['time'].axis = 'T'
-
-    # Write config settings
-    for key in self._config:
-        value = self.get_config(key)
-        if isinstance(value, (bool, type(None))):
-            value = str(value)
-        self.outfile.setncattr('config_' + key, value)
-
-    # Write additionaly metadata attributes, if given
-    if hasattr(self, 'metadata_dict'):
-        for key, value in self.metadata_dict.items():
-            self.outfile.setncattr(key, str(value))
-
-    # Add all element properties as variables
-    for prop in self.history.dtype.fields:
-        if prop in skip_parameters:
-            continue
-        # Note: Should use 'f8' if 'f4' is not accurate enough,
-        #       at expense of larger files
-        try:
-            dtype = self.history.dtype[prop]
-        except:
-            dtype = 'f4'
-        if np.issubdtype(dtype, np.integer):
-            fill=-999
-        else:
-            fill = np.nan
-        var = self.outfile.createVariable(prop, dtype, ('trajectory', 'time'), fill_value=fill)
-        for subprop in self.history_metadata[prop].items():
-            if subprop[0] not in ['dtype', 'constant', 'default', 'seed']:
-                # Apparently axis attribute shall not be given for lon and lat:
-                if prop in ['lon', 'lat'] and subprop[0] == 'axis':
-                    continue
-                var.setncattr(subprop[0], subprop[1])
+    if os.path.exists(self.outfile_name):
+        logger.warning(f'Deleting existing {self.outfile_name}')
+        os.remove(self.outfile_name)
```

Comment:
Better to raise an error? Unless user supplies a force/overwrite flag?

Reply:
Overwrite was always the default. The above might be unnecessary, basically a debug comment to show that initialisation of file is done.

Before, io_init() created the netCDF-file with dimensions and attributes etc, but no nothing else then the above happens at init, and the first write is a single call to `self.result.to_netcdf()`
Thus letting Xarray do the job of laying out the netCDF file structure.

Reply:
Ok. I guess this is how netcdf works as well.